### ISSUE TITLE
Prevent creating issues from exceptions in some situations

### ISSUE THREAD

**Issue by izuzak** [atom/notifications#37]
Wondering if we would consider preventing issues from being created in some situations in order to help users resolve problems without opening issues.

For example, we could prevent issues being created if the user is using an older version of Atom or an out-of-date package which caused the error (example: [this user](https://github.com/atom/git-diff/issues/48) had a super old version of a package installed locally) and instead tell them to update (either Atom or the package). We already [do that for non-core packages](https://github.com/atom/notifications/blob/525e2bc8c48ad42b6e3705395e6382bfbf35ead3/lib/notification-element.coffee#L139-L151) as far as I can tell, so wondering if there's a reason we wouldn't want to do that for atom/atom or core packages.

Another thing we might require is for the user to type at least a few steps which caused the issues, so that we reduce the number of issues without context.

Finally, we might consider preventing issues being created in atom/atom and atom-owned packages if the user has any non-atom-owned packages installed (and tell them to run Atom in safe mode if they want to help debug this). This again might help users figure out problems before creating issues in atom/atom, and if a problem is caused by a package -- then an issue created in atom/atom or an atom-owned package normally ends up getting closed and moved to a different package.

There's a tradeoff here -- we might not get some legit reports, but we'd also reduce the number of issues being created for non-problems and also help users help themselves. I'm thinking that if something is a legit bug -- there will be someone on the latest version willing to run Atom in safe mode and try to reproduce.

cc @benogle who's interested in these things :raised_hands:

**Comment by benogle** [comment]
Yes. I have slowly been adding better messages for errors we see a lot. 

> we could prevent issues being created if the user is using an older version of Atom

I'd love to do this. I wanted to prompt to upgrade atom if it was out of date just like the package update. I havent yet, though, because it isnt super straightforward to get info on whether atom is up to date. 

> example: this user had a super old version of a package installed locally

It might be interesting to check to see if atom packages are installed locally, then display a nice message about it.

> Another thing we might require is for the user to type at least a few steps which caused the issues, so that we reduce the number of issues without context.

This would be awesome. I wonder what this would look like design-wise. Maybe we could display a box when they click the create issue button and they can only create the issue after they enter steps

> Finally, we might consider preventing issues being created in atom/atom and atom-owned packages if the user has any non-atom-owned packages installed (and tell them to run Atom in safe mode if they want to help debug this).

I'm on the fence about this. There are some errors that dont necessarily show the package. https://github.com/atom/atom/issues/5472 is a good example. If we just forced him to be in safe mode, he would never be able to report the error, and wouldnt know what it was from. Maybe instead, we just encourage safe mode in that situation somehow?

Some other things i've seen recently that could be addressed

* Error in a package when the package dir is not in the trace. e.g. https://github.com/atom/atom/issues/5456 Often the package is in the command history, we should check for non-core packages in the command history.
* Error in a package that is seemingly disabled / uninstalled e.g. https://github.com/atom/atom/issues/5443 This is likely an issue in the package's deactivate method. It's not cleaning up after itself properly. Not sure how to fix. We _could_ see if something in the stack is coming from `.atom/packages`, and guess from there.
